exchanges/bitstamp/bitstamp_websocket.go

```diff
@@ -344,3 +413,58 @@ func (b *Bitstamp) seedOrderBook(ctx context.Context) error {
 	}
 	return nil
 }
+
+// fetchWSAuth Retrieves a userID and auth-token from REST for subscribing to a websocket channel
+// The token life-expectancy is only about 60s; use it immediately and do not store it
+func (b *Bitstamp) fetchWSAuth(ctx context.Context) (*websocketAuthResponse, error) {
+	resp := &websocketAuthResponse{}
+	err := b.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, bitstampAPIWSAuthToken, true, nil, resp)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching auth token: %w", err)
+	}
+	return resp, nil
+}
+
+// parseChannel splits the ws response channel and sets the channel type and pair
+func (b *Bitstamp) parseChannelName(r *websocketResponse) error {
+	if r.Channel == "" {
+		return nil
+	}
+
+	chanName := r.Channel
+	authParts := strings.Split(r.Channel, "-")
+	switch len(authParts) {
+	case 1:
+		// Not an auth channel
+	case 3:
+		chanName = authParts[1]
```

Comment:
Technically the pair is not required for the subscription:
```
[DEBUG] | WEBSOCKET | 07/09/2023 09:40:29 | Bitstamp websocket connection: message received: {"event":"bts:subscription_succeeded","channel":"private-my_orders-3215483","data":{}}
[ERROR] | WEBSOCKET | 07/09/2023 09:40:29 | exchange Bitstamp websocket error - unable to parse currency pair from wsResponse.Channel: channel name does not contain exactly 2 underscores: private-my_orders-3215483
[DEBUG] | WEBSOCKET | 07/09/2023 09:40:29 | Bitstamp websocket connection: message received: {"event":"bts:subscription_succeeded","channel":"private-my_trades-3215483","data":{}}
```
While I prefer the way you've setup the subscriptions in code to do it with a currencypair, a user is still able to subscribe to endpoints on their own and this will error out

Reply:
Agreed. I'll fix this and add a test.

Also: Yet another undocumented feature. It [explicitly says you have to supply currencyPair](https://www.bitstamp.net/websocket/v2/).

Reply:
Oh you don't really need to worry then. I just saw that it worked and didn't check that it wasn't supported sorry

Reply:
Oh, yeah.... cos if we can't get the pair from the channel name, it's not provided in any other way.

Think I'll sniff this a bit more. Want to see what we get back with this sub...

Reply:
I'll investigate it separately from this PR, actually.